examples/jsm/loaders/GLTFLoader.js

```diff
@@ -2966,6 +2966,15 @@ class GLTFParser {
 
 			}
 
+			texture.userData.mimeType = sourceDef.mimeType;
+
+			if ( ! sourceDef.mimeType ) {
+
+				const ext = sourceDef.uri.split( '.' ).pop().toLowerCase();
+				texture.userData.mimeType = ( ext.includes( 'jp' ) ) ? 'image/jpeg' : ext === 'webp' ? 'image/webp' : 'image/png';
+
+			}
+
```

Comment:
I think this deserves to be its own function:

```js
function getImageURIMimeType( uri ) {

	if ( uri.search( /\.jpe?g($|\?)/i ) > 0 || uri.search( /^data\:image\/jpeg/ ) === 0 ) return 'image/jpeg';
	if ( uri.search( /\.webp($|\?)/i ) > 0 || uri.search( /^data\:image\/webp/ ) === 0 ) return 'image/webp';

	return 'image/png';

}
```

And then is just this:

```js
texture.userData.mimeType = sourceDef.mimeType || getImageURIMimeType( sourceDef.uri );
```

Reply:
done.